### PR TITLE
adds sourcemap support

### PR DESCRIPTION
**Pull request by hsingh23**
Merged all my commits into a single commit that adds sourcemaps

### CODE REVIEW

**Comment by labaneilers** [comment]
Looks like the Travis CI build failed on a jshint error. Please take a look- it looks like a legit bug.

**Comment by labaneilers** [comment]
There don't appear to be any new unit tests covering the new sourcemap functionality. Can you please add one?

**Comment by hsingh23** [comment]
Yup adding tests, but I'm not sure how to run the existing tests - is there a test runner? What's the command?

**Comment by labaneilers** [comment]
Tests are executed in the default grunt task (the task is "mochaTest")

File: lib/hashly.js [hunk 1]
```diff
@@ -143,31 +146,48 @@ var processEntry = function (fullPath, baseDir, targetDir, data) {
...
+            var transformedText = null,
+                hashCode = null;
+            if (_options.processMap && fsutil.existsSync(mapFullPath)) {
+                var mapEntry = createAndAugmentManifestEntry(mapFullPath, baseDir, targetDir, data);
+                fsutil.copySync(mapFullPath, mapEntry.hashedPathPhysical);
+                if (_options.logger) {
+                    _options.logger(mapFullPath + " > " + mapEntry.hashedPathPhysical);
+                }
+                // The sourcemap filename has changed, so update the minified js/css file and replace unhashed filename with hashed one
+                // This is done to support zero downtime deployment when two different releases are out at once
```
Comment (labaneilers) [inline lib/hashly.js]: This comment seems strangely general to be applied to this line- the whole point of hashly is to support zero-downtime deployments.

**Pull request closed without merging**